src/main/java/com/hcl/appscan/jenkins/plugin/builders/AppScanBuildStep.java

```diff
@@ -295,7 +300,27 @@ private void perform(Run<?,?> build, Launcher launcher, TaskListener listener) t
     	final IProgress progress = new ScanProgress(listener);
     	final boolean suspend = m_wait;
     	final IScan scan = ScanFactory.createScan(getScanProperties(build, listener), progress, m_authProvider);
-        
+		JenkinsAuthenticationProvider checkAppScan360Connection = new JenkinsAuthenticationProvider(m_credentials,build.getParent().getParent());
+        if(m_type.equals("Dynamic Analyzer") && checkAppScan360Connection.isAppScan360()){
+			throw new AbortException(Messages.error_dynamic_analyzer_AppScan360());
+		}
+
+        if(!checkAppScan360Connection.isAppScan360()){
```

Comment:
A few things about this code:

1. Is there a reason we're checking if the connection is to AppScan 360 using one instance of JenkinsAuthenticationProvider and checking the acceptsInvalidCerts value using another instance?
2. Rather than having nested "if" statements, the 2 checks should be done as an AND check.
3. When creating Message objects, you should use the enum value rather than an int, so it's clear what type of message this is. So instead of `new Message(1, Messages.....)` you should use `new Message(Message.WARNING, Messages.....)`.

Reply:
1. I don't want to add unnecessary methods in the AuthenticationProvider class, As with that we have to make changes in the SDK.
2.  Okay, sure.
3. Okay, done.